test/Altinn.App.Core.Tests/Features/Maskinporten/Delegates/MaskinportenDelegatingHandlerTest.cs

```diff
@@ -0,0 +1,63 @@
+using Altinn.App.Core.Features.Maskinporten.Exceptions;
+using Altinn.App.Core.Features.Maskinporten.Models;
+using FluentAssertions;
+using Moq;
+
+namespace Altinn.App.Core.Tests.Features.Maskinporten.Tests.Delegates;
+
+public class MaskinportenDelegatingHandlerTest
+{
+    [Fact]
+    public async Task SendAsync_AddsAuthorizationHeader()
+    {
+        // Arrange
+        var scopes = new[] { "scope1", "scope2" };
+        var (client, handler) = TestHelpers.MockMaskinportenDelegatingHandlerFactory(
+            scopes,
+            new MaskinportenTokenResponse
+            {
+                TokenType = "Bearer",
+                Scope = "-",
+                AccessToken = "jwt-content-placeholder",
+                ExpiresIn = -1
+            }
+        );
+        var httpClient = new HttpClient(handler);
```

Comment:
Some instances here of disposables not being disposed. Adding a `using` e.g. `using var httpClient = new HttpClient(handler);` fixes this.

Reply:
I was just going with the flow based on various docs and posts, like this: https://stackoverflow.com/questions/50912160/should-httpclient-instances-created-by-httpclientfactory-be-disposed

Essentially the httprequesthandler is what you normally want to dispose, but in this case, the factory will ensure its _not_ disposed (on purpose, because of our middleware).

So I don't really know if we should dispose the client here. Chatter online indicates leave it alone, but feel free to make a firm recommendation here if you have an opinion after reading this response :)

Reply:
Can't argue with famous actor Chris Pratt.

Reply:
I dug a bit deeper into this as I have disposed httpClients before and do it instinctively.[ Microsoft does it in their example code](https://learn.microsoft.com/en-us/dotnet/core/extensions/httpclient-factory#basic-usage), but they do not explain why..